contribute/design/1.x/cstor/cstor-status.md

```diff
@@ -0,0 +1,220 @@
+---
+title: Status for CStor CRs
+authors:
+  - "@shubham14bajpai"
+owners:
+  - "@vishnuitta"
+  - "@kmova"
+  - "@AmitKumarDas"
+editor: "@shubham14bajpai"
+creation-date: 2020-02-24
+last-updated: 2020-02-24
+status: Implementable
+---
+
+# Status for CStor CRs
+
+## Table of Contents
+
+- [Status for CStor CRs](#pod-disruption-budget)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+    - [Proposed Implementation](#proposed-implementation)
+      - [CSPI Status](#cspi-status)
+      - [CSPC Status](#cspc-status)
+
+## Summary
+
+This proposal includes design details of the status information provided by the CStor CRs. The status information is used to determine the state of the corresponding CR.
+
+## Motivation
+
+The current CStor CRs don't provide appropriate information about the state of the CR and the reason for that state.
+
+### Goals
+  - Make CSPC and CSPI status informative enough to know the current state of the CR and the reason for the given state.
+
+## Proposal
+
+### User Stories
+
+As a OpenEBS user, the information regarding a CR should be present in the status or events of the CR itself rather than looking for it in controller logs.
+
+### Proposed Implementation
+
+Kuberentes provides a framework for status conditions and events which can be used to pass relevant information from the controllers to the CRs. 
+
+#### CSPI Status
+
+The current CSPI status gives us the following information via kubectl get command which are the zpool properties taken from the zfs commands and parsed to cspi status.
+
+```diff
+mayadata:maya$ kubectl -n openebs get cspi
+-  NAME                  HOSTNAME    ALLOCATED   FREE    CAPACITY  STATUS   AGE
+-  sparse-pool-1-7xq5    127.0.0.1     218K      9.94G    9.94G    ONLINE   134m
++  NAME                  HOSTNAME    ALLOCATED   FREE    CAPACITY  HEALTHYVOLUMES  STATUS   AGE
++  sparse-pool-1-7xq5    127.0.0.1     218K      9.94G    9.94G         3/5        ONLINE   134m
```

Comment:
I don't see the change in Status of CSPI to store healthyvolumes count.. don't we need it?

Reply:
We will get the list of all the volumedatasets and their status using `zfs stats` so it is not costly call we can do that.

Reply:
@vishnuitta Here we are getting `HEALTHYVOLUMES` Information from `zfs` instead of CVR which will be more appropriate.